agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.h

```diff
@@ -29,9 +30,9 @@ namespace backhaul_manager {
 class main_thread : public btl::transport_socket_thread {
 
 public:
-    main_thread(std::string temp_path_, std::set<std::string> slave_ap_ifaces_,
-                std::set<std::string> slave_sta_ifaces_, int stop_on_failure_attempts,
-                std::string const_bh_slave_);
+    main_thread(const config_file::sConfigSlave &config,
+                const std::set<std::string> &slave_ap_ifaces_,
+                const std::set<std::string> &slave_sta_ifaces_);
```

Comment:
Since you're storing references to the arguments in the main_thread object, I think there should be a (doxygen) comment that clarifies that the lifetime of those arguments must surpass that of the main_thread object itself.